src/mlpack/methods/linear_svm/linear_svm_main.cpp

```diff
@@ -0,0 +1,440 @@
+    PRINT_DATASET("predictions") + "', the following command may be used: "
+    "\n\n" +
+    PRINT_CALL("linear_svm", "input_model", "lsvm_model", "test", "test",
+        "predictions", "predictions"),
+    SEE_ALSO("@random_forest", "#random_forest"),
+    SEE_ALSO("@logistic_regression", "#logistic_regression"),
+    SEE_ALSO("LinearSVM on Wikipedia",
+        "https://en.wikipedia.org/wiki/Support-vector_machine"),
+    SEE_ALSO("mlpack::svm::LinearSVM C++ class documentation",
+        "@doxygen/classmlpack_1_1svm_1_1LinearSVM.html"));
+
+// Training parameters.
+PARAM_MATRIX_IN("training", "A matrix containing the training set (the matrix "
+    "of predictors, X).", "t");
+PARAM_UROW_IN("labels", "A matrix containing labels (0 or 1) for the points "
+    "in the training set (y).", "l");
+
+// Optimizer parameters.
+PARAM_DOUBLE_IN("lambda", "L2-regularization parameter for training.", "L",
+    0.0001);
+PARAM_DOUBLE_IN("delta", "Margin of difference between correct class and other "
+    "classes.", "d", 1.0);
+PARAM_INT_IN("num_classes", "Number of classes for classification; if "
+    "unspecified (or 0), the number of classes found in the labels will be "
+    "used.", "c", 0);
+PARAM_FLAG("no_intercept", "Do not add the intercept term to the model.", "N");
+PARAM_STRING_IN("optimizer", "Optimizer to use for training ('lbfgs' or "
+    "'psgd').", "O", "lbfgs");
+PARAM_DOUBLE_IN("tolerance", "Convergence tolerance for optimizer.", "e",
+    1e-10);
+PARAM_INT_IN("max_iterations", "Maximum iterations for optimizer (0 indicates "
+    "no limit).", "n", 10000);
+PARAM_DOUBLE_IN("step_size", "Step size for parallel SGD optimizer.",
+    "s", 0.01);
+PARAM_FLAG("shuffle", "Don't shuffle the order in which data points are "
+    "visited for parallel SGD.", "S");
+PARAM_INT_IN("epochs", "Maximum number of full epochs over dataset for "
+    "psgd", "E", 50);
+PARAM_INT_IN("seed", "Random seed.  If 0, 'std::time(NULL)' is used.", "r", 0);
+
+class LinearSVMModel
+{
+ public:
+  arma::Col<size_t> mappings;
+  LinearSVM<> svm;
+
+  template<typename Archive>
+  void serialize(Archive& ar, const unsigned int /* version */)
+  {
+    ar & BOOST_SERIALIZATION_NVP(mappings);
+    ar & BOOST_SERIALIZATION_NVP(svm);
+  }
+};
+
+
+// Model loading/saving.
+PARAM_MODEL_IN(LinearSVMModel, "input_model", "Existing model "
+    "(parameters).", "m");
+PARAM_MODEL_OUT(LinearSVMModel, "output_model", "Output for trained "
+    "linear svm model.", "M");
+
+// Testing.
+PARAM_MATRIX_IN("test", "Matrix containing test dataset.", "T");
+PARAM_UROW_IN("test_labels", "Matrix containing test labels.", "A");
+PARAM_UROW_OUT("predictions", "If test data is specified, this matrix is where "
+    "the predictions for the test set will be saved.", "P");
+PARAM_MATRIX_OUT("score", "If test data is specified, this "
+    "matrix is where the class score for the test set will be saved.",
+    "p");
```

Comment:
In other programs this is called `probabilities`; can we change the name of this to match?

Reply:
Updated.